core/templates/components/summary-tile/learner-story-summary-tile.component.ts

```diff
@@ -24,34 +24,57 @@ import { downgradeComponent } from '@angular/upgrade/static';
 import { AssetsBackendApiService } from 'services/assets-backend-api.service';
 import { AppConstants } from 'app.constants';
 import { StorySummary } from 'domain/story/story-summary.model';
+import { UrlService } from 'services/contextual/url.service';
 
 @Component({
   selector: 'oppia-learner-story-summary-tile',
   templateUrl: 'learner-story-summary-tile.component.html'
 })
 export class LearnerStorySummaryTileComponent implements OnInit {
   @Input() storySummary: StorySummary;
+  @Input() displayArea: string;
   @Input() topicName?: string;
   nodeCount: number;
   completedNodeCount: number;
   storyProgress: number;
   thumbnailUrl: string = null;
   storyLink: string;
   storyTitle: string;
+  nextIncompleteNodeTitle: string;
   storyCompleted: boolean = false;
   thumbnailBgColor: string;
   starImageUrl: string = '';
 
   constructor(
     private urlInterpolationService: UrlInterpolationService,
-    private assetsBackendApiService: AssetsBackendApiService
+    private assetsBackendApiService: AssetsBackendApiService,
+    private urlService: UrlService,
   ) {}
 
   getStoryLink(): string {
     if (!this.storySummary.getClassroomUrlFragment() ||
       !this.storySummary.getTopicUrlFragment()) {
       return '#';
     }
+    if (this.isDisplayAreaHome()) {
+      var allNodes = this.storySummary.getAllNodes();
+      var node = allNodes[this.completedNodeCount];
+      if (node) {
+        let result = '/explore/' + node.getExplorationId();
+        result = this.urlService.addField(
+          result, 'topic_url_fragment',
+          this.storySummary.getTopicUrlFragment());
+        result = this.urlService.addField(
+          result, 'classroom_url_fragment',
+          this.storySummary.getClassroomUrlFragment());
+        result = this.urlService.addField(
+          result, 'story_url_fragment',
+          this.storySummary.getUrlFragment());
+        result = this.urlService.addField(
+          result, 'node_id', node.getId());
```

Comment:
Is it possible to use some function in urlInterpolationService for this?
Also just to check:
- If there are incomplete chapters, the next incomplete chapter is started.
- If not, the story page is opened?

Reply:
I don't think we can use any function inside urlInterpolationService for this. Yes, if there are incomplete chapters, the next incomplete chapter is started. Also, if there are no incomplete chapters, the story will be considered completed and in turn topic will be completed. So, the card will be removed from this section.